File: sdks/go/pkg/models/operations/postfilesidmetadataglobalboxskillscards.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type PostFilesIDMetadataGlobalBoxSkillsCardsPathParams struct {
	FileID string `pathParam:"style=simple,explode=false,name=file_id"`
}

type PostFilesIDMetadataGlobalBoxSkillsCardsRequestBody struct {
	Cards []interface{} `json:"cards"`
}

type PostFilesIDMetadataGlobalBoxSkillsCardsRequest struct {
	Retries    *utils.RetryConfig
	PathParams PostFilesIDMetadataGlobalBoxSkillsCardsPathParams
	Request    *PostFilesIDMetadataGlobalBoxSkillsCardsRequestBody `request:"mediaType=application/json"`
}

type PostFilesIDMetadataGlobalBoxSkillsCardsResponse struct {
	ClientError        *shared.ClientError
	ContentType        string
	SkillCardsMetadata *shared.SkillCardsMetadata
	StatusCode         int64
}

File: ts-sdk/src/sdk/models/operations/postfilesidmetadataglobalboxskillscards.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class PostFilesIdMetadataGlobalBoxSkillsCardsPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=file_id" })
  fileId: string;
}


export class PostFilesIdMetadataGlobalBoxSkillsCardsRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=cards" })
  cards: any[];
}


export class PostFilesIdMetadataGlobalBoxSkillsCardsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: PostFilesIdMetadataGlobalBoxSkillsCardsPathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: PostFilesIdMetadataGlobalBoxSkillsCardsRequestBody;
}


export class PostFilesIdMetadataGlobalBoxSkillsCardsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  skillCardsMetadata?: shared.SkillCardsMetadata;

  @SpeakeasyMetadata()
  statusCode: number;
}
